### PR TITLE
EVG-16311 Add warning when using expansions in script

### PR DESCRIPTION
**Pull request by bynn**
[EVG-16311](https://jira.mongodb.org/browse/EVG-16311)

### Description 
Added a task debug log if the script was changed after expanding 

### Testing 
https://spruce-staging.corp.mongodb.com/task/sandbox_ubuntu1604_bynntask_patch_b14644c4918e0a8f632efe00e4d4e8ccc344b327_624cb29db237365d2511851e_22_04_05_21_20_48/logs?execution=1

### CODE REVIEW

File: agent/command/shell.go [hunk 1]
```diff
@@ -117,10 +117,14 @@ func (c *shellExec) Execute(ctx context.Context, _ client.Communicator, logger c
 	logger.Execution().Debug("Preparing script...")
 
 	var err error
+	var originalScript string
 	if err = c.doExpansions(conf.Expansions); err != nil {
 		logger.Execution().Warning(err.Error())
 		return errors.WithStack(err)
 	}
+	if originalScript != c.Script {
+		logger.Task().Warning("`${SHELL_EXPANSION}` syntax will not work due to evergreen expansions. See Evergreen FAQ for details.")
```
Comment (ablack12) [inline agent/command/shell.go:126]: maybe "will only work for Evergreen expansions" since that could be what users are expecting?

Comment (ablack12) [inline agent/command/shell.go:126]: .Info might be better since Warning could be a bit in your face

Comment (ablack12) [reply agent/command/shell.go]: This doesn't look updated yet to me

Comment (ablack12) [reply agent/command/shell.go]: This doesn't look updated to me

**Review by ablack12 — APPROVED** [review]
LGTM! (might need to tweak agent version again though)

File: config.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ var (
 	ClientVersion = "2022-03-11"
 
 	// Agent version to control agent rollover.
-	AgentVersion = "2022-04-04"
+	AgentVersion = "2022-04-06"
```
Comment (bynn) [inline config.go:39]:
```suggestion
	AgentVersion = "2022-04-12"
```

**Pull request merged**